kafka-client/src/main/java/dev/responsive/kafka/clients/EndOffsetsPoller.java

```diff
@@ -119,19 +121,19 @@ private synchronized Collection<Listener> getThreadMetrics() {
   }
 
   private void initPoller() {
-    LOG.info("init end offsets poller");
+    LOG.info("Initializing end offsets poller");
     if (poller != null) {
-      throw new IllegalStateException("poller already initialized");
+      throw new IllegalStateException("Poller was already initialized");
     }
-    poller = new Poller(factories.createAdminClient(configs), executor, this::getThreadMetrics);
+    poller = new Poller(() -> factories.createAdminClient(configs), executor, this::getThreadMetrics);
```

Comment:
This isn't really necessary, but I feel like having the `Poller` be responsible for creating the admin client helps  make it clear that the `Poller` is also responsible for closing the admin during shutdown. That's the only motivation behind this change